librz/util/str.c

```diff
@@ -3494,20 +3495,21 @@ RZ_API wchar_t *rz_str_mb_to_wc_l(const char *buf, int len) {
 
 RZ_API char *rz_str_wc_to_mb_l(const wchar_t *buf, int len) {
 	char *res_buf = NULL;
+	mbstate_t mbstate = { 0 };
 	size_t sz;
 
 	if (!buf || len <= 0) {
 		return NULL;
 	}
-	sz = wcstombs(NULL, buf, 0);
+	sz = wcsrtombs(NULL, &buf, 0, &mbstate);
```

Comment:
Ok, so according to https://www.cplusplus.com/reference/cwchar/wcsrtombs/ you should use `mbrlen` to initialize `mbstate`. Unfortunately i never used wcsrtombs so i would wait for @karliss to comment on this.

Reply:
https://www.gnu.org/software/libc/manual/html_node/Keeping-the-state.html says there are more than one ways to initialize `mbstate` and `memset` is one of them so `mbstate = { 0 };` should work too.